library/core/src/num/nonzero.rs

```diff
@@ -128,13 +225,15 @@ macro_rules! nonzero_integer {
         ///
         /// [null pointer optimization]: crate::option#representation
         #[$stability]
-        #[derive(Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+        #[derive(Copy, Eq)]
         #[repr(transparent)]
         #[rustc_layout_scalar_valid_range_start(1)]
         #[rustc_nonnull_optimization_guaranteed]
         #[rustc_diagnostic_item = stringify!($Ty)]
         pub struct $Ty($Int);
 
+        impl_nonzero_traits!(#[$stability] $Ty);
```

Comment:
I think having this as a separate macro call isn't valuable. I guess it provides some grouping, but not in a meaningful way, because who's to say what nonzero trait impls belong in `impl_nonzero_traits` vs elsewhere? For example the BitOr and FromStr and Display traits which are somewhere else.

I would recommend pasting what's currently in the body of `impl_nonzero_traits` directly here (or below the `impl $Ty {…}` block) without having it behind a separate macro.

Reply:
Yes, makes sense, I have moved them into this macro now.